sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs

```diff
@@ -137,7 +137,8 @@ private static void CheckTypeName(SyntaxNodeAnalysisContext context)
                 return;
             }
 
-            if (IsTypeComRelated(symbol.ContainingType) ||
+            if (string.IsNullOrWhiteSpace(identifier.ValueText) ||

```

Comment:
User is complaining about `IndexOutOfRangeException` that I cannot reproduce. This seems to be the only possible case so I am handling it here.